### PR TITLE
fix for pr590 wrong parameter for a user provider network

### PR DESCRIPTION
**Pull request by pramchan**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: controllers/openstackcluster_controller.go [hunk 1]
```diff
@@ -297,7 +297,10 @@ func (r *OpenStackClusterReconciler) reconcileNetworkComponents(log logr.Logger,
 
 		netOpts := networks.ListOpts(openStackCluster.Spec.Network)
 		networkList, err := networkingService.GetNetworksByFilter(&netOpts)
-		if err != nil && len(networkList) == 0 {
```
Comment (jichenjc) [inline controllers/openstackcluster_controller.go:300]: does this change really fix the nil reference in the bug?

**Comment by jichenjc** [comment]
you need sign cncf-cla?

File: controllers/openstackcluster_controller.go [hunk 2]
```diff
@@ -297,7 +297,10 @@ func (r *OpenStackClusterReconciler) reconcileNetworkComponents(log logr.Logger,
...
+		if len(networkList) == 0 {
+			return errors.Errorf("failed to find any network: %v", err)
+		}
+		if err != nil {
```
Comment (hidekazuna) [inline controllers/openstackcluster_controller.go:303]: It is better to check err is nil or not before checking networkList == 0.

**Comment by hidekazuna** [comment]
@pramchan Please update 

> Fixes #

to

Fixes #590

**Comment by pramchan** [comment]
Modified code to address the changes suggested...
<<<<<<< Updated upstream
                if len(networkList) == 0 {
                        return errors.Errorf("failed to find any network: %v", err)
                }
=======
>>>>>>> Stashed changes
                if err != nil {
                        return errors.Errorf("failed to find network: %v", err)
                }
                if len(networkList) <= 0 {
                        return errors.Errorf("failed to find any network: %v", err)
                }
                if len(networkList) > 1 {
                        return errors.Errorf("failed to find only one network (result: %v): %v", networkList, err)
                }

**Comment by pramchan** [comment]
Do I need to create a new pull again for  some changes?
https://github.com/pramchan/cluster-api-provider-openstack

**Comment by ncdc** [comment]
@pramchan you generally never need to create a new pull request

**Comment by hidekazuna** [comment]
> Do I need to create a new pull again for some changes?
> https://github.com/pramchan/cluster-api-provider-openstack

No, you do not need to create a new PR generally. But your PR now includes the code of some PRs already merged. https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/594/files,  so
if current situation is complicated for you, close this PR and create a new PR by following https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md.

**Comment by hidekazuna** [comment]
This PR was replaced by #610 
/close

**Pull request closed without merging**